src/core/ddsc/include/dds/ddsc/dds_public_qos.h

```diff
@@ -680,6 +746,68 @@ dds_qget_ignorelocal (
   const dds_qos_t * __restrict qos,
   dds_ignorelocal_kind_t *ignore);
 
+/**
+ * @brief Gets the names of the properties from a qos structure.
```

Comment:
The likelihood of mishaps is probably lower if it simply returns (0, ``NULL``) when the list is empty or missing, rather than leaving the out parameters untouched when it is missing.

Reply:
Fixed